### ISSUE TITLE
[1.3.0] Constructor overloads are incorrectly formatted

### ISSUE THREAD

**Issue by Bencodes** [pinterest/ktlint#2690]
## Expected Behavior

KtLint should retain the top level constructor `()` when an overloaded constructor exists and calls `this()`

## Observed Behavior

KtLint removes the default empty class level constructor when a secondary overloaded constructor exists.

## Steps to Reproduce

Given:
```kotlin
class Foo() {
    constructor(foo: String) : this() {
        // N/A
    }
}
```

KtLint produces the following Kotlin class where the default class level constructor is removed which breaks the overloaded constructors ability to call `this()`

```kotlin
class Foo {
    constructor(foo: String) : this() {
        // N/A
    }
}
```

## Your Environment

* Version of ktlint used: 1.3.0
* Relevant parts of the `.editorconfig` settings: I'm able to reproduce with an empty editor config.

**Comment by paul-dingemans** [comment]
Nice find. An empty VALUE_PARAMETER_LIST should only be disallowed in case the class does not contain a secondary constructor.
